### PR TITLE
Basic Mob Refactor and Bad Pattern Avoidance: Removes Temp and Atmos and Sentience type vars, enforces considering them

### LINKED ISSUE #37793: Some mobs are immune to vaccum, i believe this is a mistake.
[Round ID]: # (If you discovered this issue from playing tgstation hosted servers:)12806 On Citadel station

[Testmerges]: # (If you believe the issue to be caused by a test merge [OOC tab -> Show Server Revision], report it in the pull request's comment section instead.)

[Reproduction]: # (Explain your issue in detail, including the steps to reproduce it. Issues without proper reproduction steps or explanation are open to being ignored/closed by maintainers.)
During an attempt to steal the HoS gun, i blew open his office from space, and entered. I was attacked, and followed into vacuu, by the spider araneus. He proceeded to follow me into space, and kill me, because i am a pleb.  But, after checking on him several times after my death, i found that despite the office being breached, he was alive and well. I believe he should have suffocated, as spiders would need air to live.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

fixes #37793

Recently, a pr #71817 attempted to fix the fact that a lot of basic mobs have accidentally been implemented without proper environmental values, but they did it in a way very antithetical to basic mobs, they made them
🤮 DEFAULT 🤮 VARIABLES 🤮 
We're trying to avoid bloating basic mobs like that, it goes against the reason we're componentizing this behavior in the first place. It could have a valid reason, basic mobs... HAVE VARIABLES, obviously, but in this case it was setting default behavior for atmos component because people were forgetting to add it. The answer to forgetting is not bloating our variables, it is forcing contributors to consider environmental factors the mob is either immune or weak to with unit testing!

- Adds unit tests to make sure you've either flagged a mob as non-temp/atmos/sentience OR you have given them the correct components.
- Componentizes sentience options, as I thought that was relevant, but I need to revisit that as it isn't copying behavior from the old right now well, letting mobs not intended to be sentienced get sentienced.

- [ ] Reconsider removing sentience defines for now or think of a different solution for flag checking

## Why It's Good For The Game

Removes needless variables for things that may or may not be on the mob at the base level of basic mobs, godobjecting the tree.

## Changelog
:cl:
refactor: Refactors environment behavior for some mobs, report any mobs that shouldn't be surviving in space but are and vice versa. Or things that shouldn't be able to be sentienced by a potion. Sorry in advance...
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
This should now fail on every mob that wasn't given consideration on their environment elements, which is intended

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by kawoppi** [comment]
image

File: code/__DEFINES/basic_mobs.dm [hunk 1]
```diff
@@ -4,4 +4,12 @@
...
 #define UNDENSIFY_ON_DEATH (1<<2)
+/// fails unit tests if the mob has no atmos requirements yet doesn't have this flag
+#define NO_ATMOS_REQUIREMENTS (1<<3)
+/// fails unit tests if the mob has no temperature limits yet doesn't have this flag
+#define NO_TEMP_SENSITIVITY (1<<4)
+/// fails unit tests if the mob has no possibility of sentience yet doesn't have this flag
+#define NO_SENTIENCE_POSSIBILITY (1<<5)
 
+/// typical atmos requirements you'd expect most mobs who have atmos requirements at all to have.
+#define BASIC_ATMOS_REQUIREMENTS string_assoc_list(list("min_oxy" = 5, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 1, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0))
```
Comment (Mothblocks) [inline code/__DEFINES/basic_mobs.dm:15]: This is a surprisingly expensive amount of work. You should make this a GLOB instead.

**Comment by Mothblocks** [comment]
youre going to want named params for all those elements that are just numbers

**Comment by tralezab** [comment]
> #37793 is old but maybe this PR can address it properly.

Sure, since every mob will at least be forced to consider environment elements

**Comment by ATH1909** [comment]
will this cause a flame war in the comments section of any pr that adds a new basic mob over whether that mob should/shouldn't have sentience?

**Comment by tralezab** [comment]
okay we're officially on the worst timeline i'll come back to this maybe during the freeze

**Pull request closed without merging**